src/OpenSearch/Client.php

```diff
@@ -1714,16 +1613,15 @@ public function createPointInTime(array $params = [])
          * $params['body']                   = (array) The document (Required)
          *
          * @param array $params Associative array of parameters
-         * @return array
+         * @return \Http\Promise\Promise|\Psr\Http\Message\ResponseInterface
```

Comment:
returning a ResponseInterface instead of parsing its body makes it impossible to document the structure of the response (defeating half of the goals of #217).
It also forces the caller to know how each endpoint is encoding its response (is it a JSON response or a plain text response ?), while the SDK could know that (this should be part of the API description)

Reply:
This is also inconsistent with all other opensearch clients I looked at, which are all providing a proper typed response instead of a raw HTTP response.

Reply:
We don't generate a typed response unfortunately. As an intermediary step, we could return a new `OpenSearchResponseInterface` to allow us to create specific model responses in future.

We will need to support BC for the current `array` return param. We can add a new flag param `bool $returnArray = TRUE` that we trigger a deprecation if `true` and remove in 3.x. That will give users a chance to switch to expecting `OpenSearchResponseInterface`.

Reply:
Returning an array shape can be described to static analyzers (this is what #217 is about).
Today, we don't actually return an array (contrary to what the phpdoc says) but a Ringphp ArrayAccess object (which is why I could not describe the array shape yet in the generated code as it would have required making Ringphp generic while it is unmaintained).

Returning an HTTP response leaves the user of the SDK responsible of knowing how an opensearch response looks like.